src/main/kotlin/assertk/assertions/inputstream.kt

```diff
@@ -0,0 +1,62 @@
+package assertk.assertions
+
+import assertk.Assert
+import assertk.assertions.support.expected
+import java.io.InputStream
+
+fun Assert<InputStream>.hasSameContentAs(other: InputStream) {
+    val buf1 = ByteArray(4096)
+    val buf2 = ByteArray(4096)
+
+    var size = 0L
+    while(true) {
+        val len1 = actual.read(buf1)
+        val len2 = other.read(buf2)
+
+        when {
+            len1<len2 -> expected("to have the same size, but actual has size ${size+len1} which is smaller then the expected stream")
```

Comment:
Think it would be more useful to consume the rest of the other stream and report both sizes. ex:
`expected("to have the same size, but expected was:${expectedSize} and actual was:${actualSize}")